### ISSUE TITLE
Reconsider proposal check failure

### ISSUE THREAD

**Issue by Martoon-00** [tqtezos/baseDAO#48]
# Clarification and motivation

Currently, our `sProposalCheck` can return only `Bool`, but for complex predicates, it provides too few information about the failure, we want to allow users returning more complex stuff.

I also think that we should not restrict users in format of the returned thing, so user should be able to fail with `(MText, ())` in one case and with `(MText, Integer)` in another (and we will additionally wrap this error into `("FAIL_PROPOSAL_CHECK", )`)

So I propose changing the `sProposalCheck` signature to 

```haskell
:: (KnownDAOError e => e : s :-> a)
-> ProposeParams mt : s :-> s
```

The check will accept a failure function that will attach `FAIL_PROPOSAL_CHECK` tag inside. It will also make sure that all errors the user it trying to throw are documented thanks to `KnownDAOError` typeclass.

# Acceptance criteria

* `sProposalCheck` is modified as described.
* These verification errors are neatly documented.

**Comment by Martoon-00** [comment]
Oh, so it looks like one of the

**Comment by pasqu4le** [comment]
Closing this as it was solved by #273 

In the end the return type of proposal checks was restricted to be only `unit`, as the `failwith` is called by the lambda itself (and is restricted to be a pair where the left side is the `"FAIL_PROPOSAL_CHECK"` string).
